backend/ringv2/ringv2.go

```diff
@@ -318,26 +334,43 @@ func (w *watcher) hasTrigger(ctx context.Context) (bool, string, error) {
 }
 
 func (w *watcher) ensureActiveTrigger(ctx context.Context) error {
-	has, next, err := w.hasTrigger(ctx)
-	if err != nil {
-		return err
-	}
-	if has || next == "" {
-		// if next == "", there are no ring items
-		return nil
-	}
-	lease, err := w.grant(ctx)
-	if err != nil {
-		return err
+	backoff := retry.ExponentialBackoff{
+		InitialDelayInterval: 10 * time.Millisecond,
+		MaxDelayInterval:     10 * time.Second,
+		Multiplier:           10,
+		Ctx:                  ctx,
 	}
-	nextValue := path.Base(next)
-	triggerOp := clientv3.OpPut(w.triggerKey(), nextValue, clientv3.WithLease(lease.ID))
-	triggerCmp := clientv3.Compare(clientv3.Version(w.triggerKey()), "=", 0)
 
-	resp, err := w.ring.client.Txn(ctx).If(triggerCmp).Then(triggerOp).Commit()
-	if !resp.Succeeded {
-		_, _ = w.ring.client.Revoke(ctx, lease.ID)
-	}
+	err := backoff.Retry(func(retry int) (bool, error) {
+		has, next, err := w.hasTrigger(ctx)
+		if err != nil {
+			logger.WithError(err).Error("retrying")
```

Comment:
I'd suggest maybe adding a bit more context to the error than `retrying`, so the operator understand that we could not get the trigger key from etcd etc.